platform/podjs/src/browserPod.ts

```diff
@@ -15,68 +15,130 @@ import * as zip from "@zip.js/zip.js";
 import endpointsJson from "../../../../polyPod-config/endpoints.json";
 import { Manifest, readManifest } from "./manifest";
 
+const DB_NAME = "polypod";
+const DB_VERSION = 1;
+const OBJECT_STORE_POLY_IN = "poly-in";
+const OBJECT_STORE_POLY_OUT = "poly-out";
+
 const NAV_FRAME_ID = "polyNavFrame";
 const NAV_DEFAULT_BACKGROUND_COLOR = "#ffffff";
 const NAV_DARK_FOREGROUND_COLOR = "#000000";
 const NAV_LIGHT_FOREGROUND_COLOR = "#ffffff";
 
-class LocalStoragePolyIn implements PolyIn {
-    private static readonly storageKey = "polyInStore";
-    private store = JSON.parse(
-        localStorage.getItem(LocalStoragePolyIn.storageKey) || "[]"
-    );
+async function migrateData(db: IDBDatabase): Promise<IDBDatabase> {
+    const polyInData = localStorage.getItem("polyInStore");
```

Comment:
Two questions here
1. Why was the definition of `storageKey` deleted? It can still be included, and used instead of a literal here.
2. I'm not totally clear what's the dataflow here. We're assuming that there's (possibly) an item in localStorage. How did it get there?

At any rate, this is supposing knowledge of these keys and possibly its semantics. I know there were already in the original code, but now that we're bumping it up, and if you consider it in scope, it would be nice to have some documentation of this.

Reply:
>     1. Why was the definition of `storageKey` deleted? It can still be included, and used instead of a literal here.

Even if it's used only referenced once (let alone in migration code, hence legacy code)?

>     2. I'm not totally clear what's the dataflow here. We're assuming that there's (possibly) an item in localStorage. How did it get there?

We're not assuming that it exists as we check that it actually does exist before doing anything with it. As for where it's coming from, isn't it clear from the function's name that this is a migration of data that have been stored in a different fashion in the past?

> At any rate, this is supposing knowledge of these keys and possibly its semantics. I know there were already in the original code, but now that we're bumping it up, and if you consider it in scope, it would be nice to have some documentation of this.

So you are asking to type this variable as `RDF.Quad[]`? What would that guard against as we don't access the objects but just store them as-is in IndexedDB?

Reply:
> > ```
> > 1. Why was the definition of `storageKey` deleted? It can still be included, and used instead of a literal here.
> > ```
> 
> Even if it's referenced only once (let alone in migration code, hence legacy code)?

Well, literals-as-constants is a pretty solid rule, in general.
> 
> > ```
> > 2. I'm not totally clear what's the dataflow here. We're assuming that there's (possibly) an item in localStorage. How did it get there?
> > ```
> 
> We're not assuming that it exists as we check that it actually does exist before doing anything with it. As for where it's coming from, isn't it clear from the function's name that this is a migration of data that have been stored in a different fashion in the past?

Someone correct me if I'm wrong, but I'm pretty sure no one has been using and storing anything in localStorage so far, there's no installed user base. We've been using it for testing purposes, mainly.
> 
> > At any rate, this is supposing knowledge of these keys and possibly its semantics. I know there were already in the original code, but now that we're bumping it up, and if you consider it in scope, it would be nice to have some documentation of this.
> 
> So you are asking to type this variable as `RDF.Quad[]`? What would that guard against as we don't access the objects here but just store them as-is in IndexedDB?

Er, no. Simply to have a single source of truth, even if it's just in documentation, for the actual keys we're using to store stuff.

Reply:
Regarding 1: Something used only once that's not some weird magic number shouldn't be a constant IMHO, but up to @snoack since he's writing the code - we don't have style rules about that kind of stuff yet.

Regarding 2: We're not using pod.js in production, but everybody is using it for local development. I think it's nice if we don't disrupt that, it just keeps on working. But we should probably add a nice `TODO` comment to the migration logic, saying that we can eliminate it after October 2022. I'm not super attached to having migration logic though, given how it's not affecting anyone other than our devs...

Reply:
I'm in the same boat as @fhd regarding inline usage of literals. However, I'm happy to adjust for consistency and compliance with any coding styles. But it seems there are no coding style rules regarding this, and as for consistency this isn't the only place where string literals are used inline in `pod.js`. Moreover, it makes the migration logic, more self-contained, hence easier to remove (without leaving artifact) as we all seem to agree to do at some point.

As for whether to remove the migration logic now, or in a few months, I don't have a strong preference. I can see @fhd's point about not disrupting the developer experience. But also if we jusst get rid of it now, that would eliminate some discussions to be resolved here otherwise, let alone that the migration logic isn't very well tested.